### ISSUE TITLE
Flutter Inspector not available in Android Studio

### ISSUE THREAD

**Issue by kenzieschmoll** [flutter/flutter-intellij#7430]
I can only access the Flutter Deep Links tool window, but not any of the other expected tool windows. This is with the new UI. The latest IntelliJ with the new UI and latest plugins is WAI, so this may be AS only.

![Screenshot 2024-05-15 at 9 30 19 AM](https://github.com/flutter/flutter-intellij/assets/43759233/c093d53e-a464-4928-894f-82599810c980)

```
Android Studio Jellyfish | 2023.3.1
Build #AI-233.14808.21.2331.11709847, built on April 12, 2024
Runtime version: 17.0.10+0-17.0.10b1087.21-11572160 x86_64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
macOS 14.4.1
GC: G1 Young Generation, G1 Old Generation
Memory: 4096M
Cores: 16
Metal Rendering is ON
Registry:
  ide.experimental.ui=true
Non-Bundled Plugins:
  Dart (233.15123)
  io.flutter (79.0.2)
```

**Comment by helin24** [comment]
Are you able to run your app when this happens? I wonder if the project isn't being recognized as a Dart/Flutter project for some reason.

**Comment by martipello** [comment]
i have the same issue, this happens everytime i update android studio, before i found and removed multiple android studio folders here 
/Users/userName/Library/Application Support/Google
i applied the fix described here 
https://github.com/flutter/flutter-intellij/issues/5853
but after the jellyfish update this didnt work anymore 🤷

**Comment by helin24** [comment]
@martipello you've been using the steps here? https://github.com/flutter/flutter-intellij/issues/5853#issuecomment-963254827

I'm not sure I see how this is related, as some of the windows here don't you JX browser at all. So for you, before you fix the location of JX browser, this removes the inspector, performance, and outline windows from your tools menu. After you fix the JX browser file, the tool windows are available?

**Comment by kenzieschmoll** [comment]
> Are you able to run your app when this happens? I wonder if the project isn't being recognized as a Dart/Flutter project for some reason.

Yes, and the Inspector and Outline are still not available when I run an app. Lmk if there are any logs you want me to collect to help debug.

**Comment by helin24** [comment]
Kenzie let me know that this can be reproduced by opening `devtools/packages`, which is a directory with multiple flutter projects within it. I've looked at our code a little more, and this issue is related to how we detect Flutter modules within an open project. I feel like there was a larger discussion about monorepos and projects that have multiple directories with their own pubspec files. @jwren are you familiar with this topic?

In the meantime if you need a workaround, opening a project with a pubspec within it should show all of the Flutter windows again.

**Comment by martipello** [comment]
> Kenzie let me know that this can be reproduced by opening `devtools/packages`, which is a directory with multiple flutter projects within it. I've looked at our code a little more, and this issue is related to how we detect Flutter modules within an open project. I feel like there was a larger discussion about monorepos and projects that have multiple directories with their own pubspec files. @jwren are you familiar with this topic?
> 
> In the meantime if you need a workaround, opening a project with a pubspec within it should show all of the Flutter windows again.

That's right I think you can create a near empty one also, my project is multi module

**Comment by helin24** [comment]
Okay, makes sense. I'm still confounded by why the JX browser steps would have fixed this issue for you though @martipello  🤷

**Comment by martipello** [comment]
I thought it fixed it for me the first time it happened I tried it again today as the latest update broke it and this time it didn't fix it

**Comment by jwren** [comment]
No update, prioritizing to take a closer look soon.

See https://github.com/flutter/flutter-intellij/issues/7875.  I current guess is that this is race condition hit by some users.  Perhaps as a consequence of having to download the JxBrowser perhaps.

**Comment by jwren** [comment]
This was a sporadic issue, I had it reproduced by myself and a co-worker earlier this week, https://github.com/flutter/flutter-intellij/issues/8029.  Testing and validating a fix as part of the M85 release.
